crates/bevy_picking/src/pointer.rs

```diff
@@ -29,41 +27,69 @@ use crate::backend::HitData;
 #[derive(Debug, Default, Clone, Copy, Eq, PartialEq, Hash, Component, Reflect)]
 #[require(PointerLocation, PointerPress, PointerInteraction)]
 #[reflect(Component, Default, Debug, Hash, PartialEq)]
-pub enum PointerId {
-    /// The mouse pointer.
-    #[default]
-    Mouse,
-    /// A touch input, usually numbered by window touch events from `winit`.
-    Touch(u64),
-    /// A custom, uniquely identified pointer. Useful for mocking inputs or implementing a software
-    /// controlled cursor.
-    #[reflect(ignore)]
-    Custom(Uuid),
+pub struct PointerId {
```

Comment:
Given that PointerId is now a common "thing" maybe it should just be `PointerId(u64)` and `PointerKind` would just be stored adjacent to it in places where that is relevant (ex: as a separate Component)? To distinguish between two pointers, don't we only need to compare / hash the ID, not the kind?

Reply:
If it is _not_ a common thing. Then I think we should move back to a typed-wrapper to protect against comparing "unlike" ids across types.

Reply:
Right now (and before this PR), the IDs are only meaningfully used by touches. In theory you could encode information for Virtual / Custom types in the u64 / UUID, but I'm not sure what you would use that for.

In either case, in order for two PointerId to match, both the kind and the number associated with it must match. As a result, I don't think we should split these up, but I'm fine to go back to an enum design here. Which do you prefer?

1. The current design, with split Id + kind.
2. `enum PointerId`, only `Touch` gets a `u64`.
3.  `enum PointerId`, each kind has a `u64`.

I think I prefer solution 2, but went with solution 1 to ease migration.

Reply:
I also prefer solution (2). I think thats the move.